cmd/rook/ceph/osd.go

```diff
@@ -116,10 +116,18 @@ func addOSDConfigFlags(command *cobra.Command) {
 func init() {
 	addOSDFlags(osdCmd)
 	addCephFlags(osdCmd)
-	flags.SetFlagsFromEnv(osdCmd.Flags(), rook.RookEnvVarPrefix)
-	flags.SetFlagsFromEnv(osdConfigCmd.Flags(), rook.RookEnvVarPrefix)
-	flags.SetFlagsFromEnv(provisionCmd.Flags(), rook.RookEnvVarPrefix)
-	flags.SetFlagsFromEnv(osdStartCmd.Flags(), rook.RookEnvVarPrefix)
+	if err := flags.SetFlagsFromEnv(osdCmd.Flags(), rook.RookEnvVarPrefix); err != nil {
```

Comment:
Instead of a panic everywhere that SetFlagsFromEnv() is called, let's change the signature of SetFlagsFromEnv() so it doesn't return an error, and then panic from inside that method.

Reply:
Done.